### ISSUE TITLE
Logging interface tap_error_log prints the wrong source crate

### ISSUE THREAD

**Issue by TheQuantumPhysicist** [mintlayer/mintlayer-core#1145]
When prints happen from tap_error_log, the source of the error is lost and the source is considered to be the utils crate itself. This is undesirable as it hides the source of problems. We should find a way to get around this.

**Comment by iljakuklic** [comment]
Horrible hacks aside, I am afraid it probably needs to be implemented as a macro to support this.

**Comment by TheQuantumPhysicist** [comment]
> Horrible hacks aside, I am afraid it probably needs to be implemented as a macro to support this.

Yep. I can see that's the only way. Though we have to at least acknowledge and study this because it's causing damage in debugging. 

Another solution, btw, is code duplication... not the best either.

**Comment by ImplOfAnImpl** [comment]
Addressed by https://github.com/mintlayer/mintlayer-core/pull/1601
